### PR TITLE
sort blog feeds by year

### PR DESCRIPTION
**Pull request by Ryuno-Ki**
Signed-off-by: André Jaenisch <andre.jaenisch@posteo.de>

Challenge accepted!

refs #81

### CODE REVIEW

File: lib/rss.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import Parser from "rss-parser";
 
-export async function getFeed() {
+export function getFeed() {
 	const parser = new Parser();
-	const feed = await parser.parseURL("https://dev.to/feed/inezabonte");
-
-	return feed;
+	return parser.parseURL("https://dev.to/feed/inezabonte");
```
Comment (Ryuno-Ki) [inline lib/rss.js:5]: `parser.parseURL` already returns a Promise. No need to await it and return a Promise of a resolved Promise ;-)

File: pages/blog.js [hunk 2]
```diff
@@ -15,8 +15,9 @@ export default function Blog({ items }) {
 					</p>
 				</div>
 				<div className="space-y-8">
-					{items.map((item) => (
-						<div key={item.link} className="space-y-2">
+					<h2 className="text-3xl font-bold dark:text-white mb-4">In 2021</h2>
```
Comment (Ryuno-Ki) [inline pages/blog.js:18]: I have no clue of Tailwind. Adjust as you see fit.

File: pages/blog.js [hunk 3]
```diff
@@ -15,8 +15,9 @@ export default function Blog({ items }) {
...
+					{items['2021'].map((item) => (
```
Comment (Ryuno-Ki) [inline pages/blog.js:19]: You could iterate on sorting by month down the road.

File: pages/blog.js [hunk 4]
```diff
@@ -15,8 +15,9 @@ export default function Blog({ items }) {
...
+						<article key={item.link} className="space-y-2">
```
Comment (Ryuno-Ki) [inline pages/blog.js:20]: Would be great if these became [`h-entry`](https://indieweb.org/article) formatted!

File: pages/blog.js [hunk 5]
```diff
@@ -35,7 +36,31 @@ export default function Blog({ items }) {
 							<p className="text-lg text-gray-600 dark:text-gray-400">
 								{format(new Date(item.isoDate), "PPP")}
 							</p>
-						</div>
+						</article>
+					))}
+
+					<h2 className="text-3xl font-bold dark:text-white mb-4">In 2020</h2>
+					{items['2020'].map((item) => (
```
Comment (Ryuno-Ki) [inline pages/blog.js:43]: Since I'm repeating myself here, you might want to break it out into a separate component.

File: pages/blog.js [hunk 6]
```diff
@@ -48,7 +73,14 @@ export async function getStaticProps() {
 
 	return {
 		props: {
-			items: detailedFeed.items,
+			items: {
+				2020: detailedFeed.items.filter((item) => {
```
Comment (Ryuno-Ki) [inline pages/blog.js:77]: Consider sorting them to be really sure.
For now, I simply filter.

**Review by inezabonte — APPROVED** [review]

File: pages/blog.js (hunk 4 above)
Comment (Ryuno-Ki) [reply pages/blog.js:20]: https://indieweb.org/article (I actually linked it)

**Comment by inezabonte** [comment]
@Ryuno-Ki . You didn't give me a cat picture

**Comment by Ryuno-Ki** [comment]
cat on a roomba

Fixed!

**Pull request merged**